packages/core/contracts/oracle/implementation/VotingV2.sol

```diff
@@ -308,49 +305,63 @@ contract VotingV2 is
+
+            // If the request status is not resolved then we are currently in the voting round assoicated with this
+            // request and cant apply any slashing to this request. Note that we dont need to worry about rolled votes
+            // here as this is deal with in the _updateCumulativeSlashingTrackers function by adding the request index
+            // to the deletedRequests and so we'd have skipped it for free from this mapping.
             if (_getRequestStatus(priceRequest, currentRoundId) != RequestStatus.Resolved) break;
 
+            VoteInstance storage voteInstance = priceRequest.voteInstances[priceRequest.lastVotingRound];
+            SlashingTracker storage requestSlashingTracker = priceRequest.slashingTrackers;
+
             bytes32 revealHash = voteInstance.voteSubmissions[voterAddress].revealHash;
             // The voter did not reveal or did not commit. Slash at noVote rate.
             if (revealHash == 0)
-                slash -= int256((voterStake.activeStake * requestSlashingTrackers[i].noVoteSlashPerToken) / 1e18);
+                slash -= int256((voterStake.activeStake * requestSlashingTracker.noVoteSlashPerToken) / 1e18);
 
                 // The voter did not vote with the majority. Slash at wrongVote rate.
             else if (!voteInstance.resultComputation.wasVoteCorrect(revealHash))
-                slash -= int256((voterStake.activeStake * requestSlashingTrackers[i].wrongVoteSlashPerToken) / 1e18);
+                slash -= int256((voterStake.activeStake * requestSlashingTracker.wrongVoteSlashPerToken) / 1e18);
 
                 // The voter voted correctly. Receive a pro-rate share of the other voters slashed amounts as a reward.
             else
                 slash += int256(
-                    (((voterStake.activeStake * requestSlashingTrackers[i].totalSlashed)) /
-                        requestSlashingTrackers[i].totalCorrectVotes)
+                    (((voterStake.activeStake * requestSlashingTracker.totalSlashed)) /
+                        requestSlashingTracker.totalCorrectVotes)
                 );
 
             // If this is not the last price request to apply and the next request in the batch is from a subsequent
             // round then apply the slashing now. Else, do nothing and apply the slashing after the loop concludes.
             // This acts to apply slashing within a round as independent actions: multiple votes within the same round
-            // should not impact each other but subsequent rounds should impact each other.
-            if (priceRequestIdsLength - i > 1 && roundId != priceRequestIds[i + 1].roundId) {
-                applySlashToVoter(slash, voterAddress);
+
+            // should not impact each other but subsequent rounds should impact each other. We need to consider the
+            // deletedRequests mapping when finding the next index as the next request may have been deleted or rolled.
+            uint256 nextRequestIndex =
```

Comment:
small downside is now the need to do this. note that there was actually a bug here before with this if the deletion happened right up to the last request this method could revert. this now saves us on this as well as working with rolled votes.